modules/auth/proc-adding-google-as-an-authentication-provider.adoc

```diff
@@ -0,0 +1,93 @@
+.. Go to to link:https://console.cloud.google.com/apis/credentials[*APIs & Services > Credentials*],
+click *Create Credentials*, choose *OAuth client ID*, and create a credential with following parameters:
+*Application type*:: Select `Web Application`.
+*Name*:: Enter your application name, such as `{product}`.
+*Authorized JavaScript origins*:: Add URI: `pass:[http://localhost:3000]`
+*Authorized redirect URIs*:: Click *Add URI* to enter the backend authentication URI set in {product}:
+For local development::: `pass:[http://localhost:7007/api/auth/google/handler/frame]`
+For a non-local deployment::: `pass:[http://{APP_FQDN}:{APP_BACKEND_PORT}/api/auth/google/handler/frame]`
+.. Save the *Client ID* and *Client secret* for the next step.
+
+
+
+. Set up the Google authentication provider in the {product} backend (`app-config.yaml`).
++
+Add the `google` provider configuration to your `app-config.yaml` under the root `auth` configuration:
++
+[source,yaml]
+----
+auth:
+  environment: development
+  providers:
+    google:
+      development:
+        clientId: ${AUTH_GOOGLE_CLIENT_ID}
+        clientSecret: ${AUTH_GOOGLE_CLIENT_SECRET}
+----
++
+The Google provider is a structure with these configuration keys:
+
+`clientId`:: The client ID that you generated, such as `10023341500512-beui241gjwwkrdkr2eh7dprewj2pp1q.apps.googleusercontent.com`
+`clientSecret`:: The client secret tied to the generated client ID, such as ``.
+
+
+. (Optionally) To enable the Google provider for sign-in,
+add the provider to the {product} front-end (`packages/app/src/App.tsx`).
++
+To add the provider to the front-end, add the `googleAuthApi` reference and
+`SignInPage` component to the sign-in page.
++
+To configure sign-in, provide a custom `SignInPage` app component.
+The app renders this component before any other routes.
+The component provides the identity of the current user.
+The `SignInPage` can render any number of pages and components, or just blank space with logic running in the background.
+In the end however it must provide a valid {product} user identity through the onSignInSuccess callback prop, at which point the rest of the app is rendered.
++
+If you want to, you can use the `SignInPage` component provided by `@backstage/core-components`, which takes either a provider or providers (array) prop of `SignInProviderConfig` definitions.
++
+The following example for Google shows the additions needed to `packages/app/src/App.tsx`:
++
+.`packages/app/src/App.tsx`
+[source,javascript]
+----
+import { googleAuthApiRef } from '@backstage/core-plugin-api';
+import { SignInPage } from '@backstage/core-components';
+const app = createApp({
+  components: {
+    SignInPage: props => (
+      <SignInPage
+        {...props}
+        auto
+        provider={{
+          id: 'google-auth-provider',
+          title: 'Google',
+          message: 'Sign in using Google',
+          apiRef: googleAuthApiRef,
+        }}
+      />
+    ),
+  },
+  // ..
+});
+----
```

Comment:
This whole section is not necessary. in RHDH we have `signInPage` option that simplifies all this.  So all that is required is to add `signInPage: google` to the configuration. 

This is partially explained in https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/76 But only partially, we need better documentation of what this option is for.